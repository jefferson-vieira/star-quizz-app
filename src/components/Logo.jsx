import React from 'react';

import logo from '../assets/img/logo.png';

const Logo = () => (
  <div id="logo" className="col-12 col-sm-10 col-md-6">
    <img id="logo-img" className="img-fluid" alt="StarQuizz Logo" src={logo} />
  </div>
);

export default Logo;
