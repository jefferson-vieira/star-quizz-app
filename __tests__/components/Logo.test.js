import React from 'react';
import { shallow, mount } from 'enzyme';

import Logo from '../../src/components/Logo';
import logo from '../../src/assets/img/logo.png';

describe('Testing Logo Component...', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Logo />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show logo', () => {
    const wrapper = mount(<Logo />);
    expect(wrapper.find('img').prop('src')).toEqual(logo);
  });
});
