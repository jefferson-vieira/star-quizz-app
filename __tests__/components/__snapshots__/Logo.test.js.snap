// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Logo Component... should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Logo />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="StarQuizz Logo"
        className="img-fluid"
        id="logo-img"
        src="logo.png"
      />,
      "className": "col-12 col-sm-10 col-md-6",
      "id": "logo",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "StarQuizz Logo",
        "className": "img-fluid",
        "id": "logo-img",
        "src": "logo.png",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="StarQuizz Logo"
          className="img-fluid"
          id="logo-img"
          src="logo.png"
        />,
        "className": "col-12 col-sm-10 col-md-6",
        "id": "logo",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "StarQuizz Logo",
          "className": "img-fluid",
          "id": "logo-img",
          "src": "logo.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
